tests/unit/executors/indexers/test_numpyindexer.py

```diff
@@ -160,3 +160,25 @@ def test__get_sorted_top_k(batch_size, num_docs, top_k):
 
         np.testing.assert_equal(idx, expected_idx)
         np.testing.assert_equal(dist, expected_dist)
+
+
+@pytest.mark.parametrize('batch_size, compress_level', [(None, 0), (None, 1), (2, 0), (2, 1)])
+def test_numpy_indexer_empty_data(batch_size, compress_level, test_metas):
+    np.random.seed(500)
+    num_dim = 64
+    num_query = 10
+    query = np.array(np.random.random([num_query, num_dim]), dtype=np.float32)
+
+    idx_file_path = os.path.join(test_metas['workspace'], 'np.test.gz')
+    with NumpyIndexer(index_filename=idx_file_path, compress_level=compress_level, metas=test_metas) as indexer:
+        indexer.batch_size = batch_size
+        indexer.touch()
+        indexer.save()
+        assert os.path.exists(indexer.index_abspath)
+        save_abspath = indexer.save_abspath
+
+    with BaseIndexer.load(save_abspath) as indexer:
+        assert isinstance(indexer, NumpyIndexer)
+        idx, dist = indexer.query(query, top_k=4)
+        assert idx.shape == dist.shape
+        assert idx.shape == (num_query, 0)
```

Comment:
so is the previous `numpy.empty` designed for complying with this?

Reply:
That's just the `.shape`, but yes.

Reply:
in #1229 you mentioned
> If a shard is empty, is should still return an empty list of results.

so the desired behavior is to return an empty ndarray with the shape `[num_query, 0]`?

Reply:
is there some reason to keep the `axis=0`? or what the choice between
```python
>>> np.empty([0, 0])
array([], shape=(0, 0), dtype=float64)
>>> np.empty([3, 0])
array([], shape=(3, 0), dtype=float64)
>>> np.empty([0])
array([], dtype=float64)
```

Reply:
> in #1229 you mentioned
> 
> > If a shard is empty, is should still return an empty list of results.
> 
> so the desired behavior is to return an empty ndarray with the shape `[num_query, 0]`?

Yes

Reply:
I think we want this, just so that the shapes still asign (nr of queries - first dim)

```python
>>> np.empty([3, 0])
array([], shape=(3, 0), dtype=float64)
```

Reply:
can we `return None, None`?

Reply:
what I don't like about this part is: as a user of this API, you don't know if the indexer is empty *until you inspect at axis=1*, and then say "oh, axis=1 is 0, so it must be empty!" - which isn't an obvious connection unless you documented in that way.

Return `None, None` would be much more direct.

Reply:
But, for example, in the case of sharding, where we want to merge results from two different shards, if one is empty, it's OK, they can merge. If one is `None, None`, error. Right?

Reply:
then this logic should be put into driver. which merge are we talking about? can you point out the function?